pkg/chunk/gcp/gcs_object_client.go

```diff
@@ -130,7 +137,7 @@ func (s *GCSObjectClient) List(ctx context.Context, prefix, delimiter string) ([
 		}
 
 		// When doing query with Delimiter, Prefix is the only field set for entries which represent synthetic "directory entries".
-		if attr.Name == "" {
+		if attr.Prefix != "" {
```

Comment:
May you confirm Prefix is not also set objects?

Reply:
It is not an attribute returned in objects (“items”) in the API. Only Go exposes it via same struct.

See “Response” in https://cloud.google.com/storage/docs/json_api/v1/objects/list

Reply:
And https://cloud.google.com/storage/docs/json_api/v1/objects for list of fields for objects.